### PR TITLE
mill: update contract compilation and execution strategy to enable `/=` for library imports

### PR DESCRIPTION
**Pull request by dr-frmr**
(WIP) 

This PR adds support for library imports in smart contracts. This upgrade will require smart-lib.noun to be recompiled along with every contract.

The contract type, `wheat`, has been changed to `[cont=(unit [bat=* pay=(list *)]) owns=(set id)]`, where `bat` is the contract itself and `pay` is a list of the nocks for each library that the contract imports. 

The `parse.hoon` generator replaces `deploy.hoon` for compiling contracts. Before merging we can replace `deploy.hoon` with it so as not to just leave it out there, plus that name is more correct. `parse.hoon` takes a contract file with optional `/=` imports and compiles it.

TODO: 
- [x] build an arm in `mill` which matches the compiling behavior and combines the nocks stored in a `wheat`, producing `crop`.
- [x] finish `parse.hoon` so it returns the full `[bat=* pay=(list *)]`
- [x] delete `deploy.hoon`, replace with `parse.hoon`
- [x] recompile smart-lib.noun and all contracts in repo, sans those being developed elsewhere

### CODE REVIEW

**Comment by dr-frmr** [comment]
Was thinking about the same problem just now. That sounds like a good way to do it, but is it concerning for performance at all that we have to recombine the libraries when we want to run a contract call?

**Comment by tacryt-socryp** [comment]
I'm not concerned (yet). Much more concerned about keeping space usage small, which this tactic will accomplish. We can always factor it into gas cost later.

**Comment by tacryt-socryp** [comment]
Also worth noting is that including large libraries will cause large amounts of hashing even if the whole library isn't used, so there will be a strong developer incentive towards having imports stay as minimal as possible.

**Comment by tacryt-socryp** [comment]
The good news is that single vs multi-file shouldn't provide a large difference to the gas cost to publish or to call a contract based on how zink works. Main difference will be total amount of nock code involved.

**Comment by tacryt-socryp** [comment]
Happy to pair here. Keep in mind that we'll need to play nice with the fact that the payload in this case (for a contract) is the payload of a door, which includes the door sample.

**Comment by dr-frmr** [comment]
I'm down to keep messing with it this weekend and will hit you up Monday if I can't get it working

**Comment by dr-frmr** [comment]
Ugh, this still isn't working right. We got the trivial contract to run without actually using anything from the libraries, but when you try and use one of their arms, we're back to failing on the `shut`.

**Comment by tacryt-socryp** [comment]
Note that we must find a way to separate full-nok into a list of * for structural sharing to work properly, so we'll need to figure out the steps necessary for that.

**Comment by dr-frmr** [comment]
Ok, I've tested the updated `mill` and recompiled `smart-lib.noun` along with contracts under my purview -- I'll send out a PSA to uqteam announcing the change and that people will need to recompile their contracts. Main change is that the .noun is now the full wheat, and you have to `;;` it when you code it into a town deployment generator such as in `init.hoon`.

I merged in `master` and commented out the `zebra` call in `weed` -- @tacryt-socryp let me know if you want it set up the other way around, I just did this so we can use master to set up testnet again with new changes and have transactions run through fast enough.

File: lib/zig/mill.hoon [hunk 1]
```diff
@@ -227,29 +229,39 @@
...
-            !>(cart)
-          !>(embryo)
-        =/  =book
-          (zebra budget cax gun)
-        ?:  ?=(%| -.p.book)
-          ~&  >>  %crashed
-          =/  res=(unit *)
-            (mack gun)
-          ~&  ?~(res %also-crashed ;;(chick u.res))
+        =/  res
```
[the remark this answers is not in the document]
Comment (dr-frmr) [reply lib/zig/mill.hoon:239]: agreed, for sure -- should I keep zink version uncommented and comment out mule version for now?

File: gen/deploy.hoon [hunk 2]
```diff
@@ -1,12 +1,80 @@
...
+=/  perfect  .*([q.smart-lib q.payload] q.cont)
+=/  dor  [-:!>(*contract:smart) perfect]
+::
 :-  %noun
-q:(slap contract (ream '-'))
\ No newline at end of file
+^-  wheat:smart
+[`[bat=q.cont pay=full-nock] ~]
+::
+++  shut                                               ::  slam a door
```
[the remark this answers is not in the document]
Comment (dr-frmr) [reply gen/deploy.hoon:29]: oh, yeah

**Pull request merged**